pkg/server/chaosd/network.go

```diff
@@ -85,6 +85,25 @@ func (networkAttack) Attack(options core.AttackConfig, env Environment) (err err
 				return errors.WithStack(err)
 			}
 		}
+
+	case core.NetworkNICDownAction:
+		if err := env.Chaos.getNICIP(attack); err != nil {
+			return errors.WithStack(err)
+		}
+
+		NICDownCommand := fmt.Sprintf("ifconfig %s down", attack.Device)
+
+		cmd := exec.Command("bash", "-c", NICDownCommand)
+		stdout, err := cmd.CombinedOutput()
+		if err != nil {
+			log.Error(cmd.String()+string(stdout), zap.Error(err))
+			return errors.WithStack(err)
+		}
+
+		if attack.Duration != "-1" {
```

Comment:
is this should be `len(attack.Duration) != 0`?

Reply:
`len(attack.Duration)` has been checked before this function, `-1` means attack will run continuously. It will be marked in doc.

Reply:
Why not check this field before executing this attack?

Reply:
This field has been checked for necessity before executing, `if len(n.Duration) == 0 {
		return errors.New("duration is required")
	}`, the judgement here is only to decide whether to run continuously.

Reply:
Actually, this `-1` option is added on purpose, because I think some user may need this option to let the experiment run continuously, especially those who run NIC down locally rather than remotely. And `attack.Duration` is hardly ever set to negative numbers, I think `-1` is enough to be a warning sign (maybe `-2333` or `-32768` is better).